### DISCUSSION TITLE
Use Stacker for Fluid Meshing

### DISCUSSION THREAD

**Discussion by kmahajan-cadfem** [ansys/pyprimemesh#653] [Q&A] [answered]
customer is using the [example](https://prime.docs.pyansys.com/version/stable/examples/gallery_examples/gallery/05_pcb_stacker.html#sphx-glr-examples-gallery-examples-gallery-05-pcb-stacker-py) to create meshing for the fluid domain using the PyPrimeMesh, following are some questions that came up while discussing the problems from customer side.

1. PrimeMesh outputs everything as a fluid. Is there a possibility that it is also saved as solid?

2. The NamedSelections are the biggest problem for me right now: Using CadReaderRoute.Workbench and Create_Zones_From_Labels I could at least find the NS in Fluent, but they are split for each part. However, I need the NS as zones as defined in SpaceClaim

3. The sweeper.stack_base_face function can set the number of elements using max-offset-size. However, I am looking for a way to set the same number of elements for each layer of the PCB. Is there also a parameter for this?

since the Prime Server is not documented and somewhat new for us, I cannot get into all the details, could you please shade some light to it, 

if this is not possible at the moment out of the box, please let  me know I will create the feature request

**Comment by waltersma** [comment, marked as answer]
Hi @kmahajan-cadfem,

Please see comments below:

1. Yes, you can set the [cell zonelet type](https://prime.docs.pyansys.com/version/stable/api/_autosummary/ansys.meshing.prime.CellZoneletType.html#ansys.meshing.prime.CellZoneletType) for each volume as a volume control. An example of how this is done can be seen [here](https://prime.docs.pyansys.com/version/stable/examples/gallery_examples/gallery/08_lucid_generic_f1_rear_wing.html#define-volume-controls).  And then use these volume controls when defining parameters for volume meshing as shown later in the same example [here](https://prime.docs.pyansys.com/version/stable/examples/gallery_examples/gallery/08_lucid_generic_f1_rear_wing.html#generate-volume-mesh).
2.  It is difficult to advise without additional information on how the model is constructed and what the exact output is intended but if I understand the request correctly you have named selections for face zonelets that span more than a single part that are split when reading into the solver and you wish to maintain them as a single zone?  If this is the case then would it be possible for you to merge the parts after volume meshing and before creating the zones?  If this is not correct then we may need more information to help answer the question as there are a few assumptions about the model you have and the methods used.
3.  There is a methodology that is outlined in a more advanced example that should be part of 0.5.0 and that is available here: [Multi-layer PCB](https://prime.docs.pyansys.com/version/dev/examples/gallery_examples/gallery/09_multi_layer_quad_mesh_pcb.html#sphx-glr-examples-gallery-examples-gallery-09-multi-layer-quad-mesh-pcb-py).  The example shows how to use edge sizing to define the number of cells through each layer.  A named selection is defined in the scdoc for the edges and is named such that the length of the edges is included in the name.  This is then used to define the number of cells through each volume.

I hope these answers help!

Thanks
